File: calculus_and_numerical_methods/taylor_series.cpp

/**
 * Taylor Series Approximations for e^x, sin(x), cos(x)
 * Author: Scott Douglass
 * License: MIT
 */

#include <iostream>
#include <cmath>

double exp_taylor(double x, int n = 10) {
    double sum = 1.0;
    double term = 1.0;
    for (int i = 1; i <= n; ++i) {
        term *= x / i;
        sum += term;
    }
    return sum;
}

int main() {
    double x = 1.0;
    std::cout << "e^1 ≈ " << exp_taylor(x) << std::endl;
    return 0;
}

File: calculus_and_numerical_methods/taylor_series.py
"""
taylor_series.py
Compute Taylor series approximation for exp(x), sin(x), cos(x).
Author: Scott
"""

import math

def taylor_exp(x, n=10):
    return sum((x**i)/math.factorial(i) for i in range(n))

def taylor_sin(x, n=10):
    return sum(((-1)**i)*(x**(2*i+1))/math.factorial(2*i+1) for i in range(n))

def taylor_cos(x, n=10):
    return sum(((-1)**i)*(x**(2*i))/math.factorial(2*i) for i in range(n))

if __name__ == "__main__":
    x = float(input("Enter x (default 1.0): ") or 1.0)
    terms = int(input("Number of terms (default 10): ") or 10)
    print(f"exp({x}) ≈ {taylor_exp(x, terms):.6f}")
    print(f"sin({x}) ≈ {taylor_sin(x, terms):.6f}")
    print(f"cos({x}) ≈ {taylor_cos(x, terms):.6f}")
